lib/scraped/response/decorator/absolute_urls.rb

```diff
@@ -15,7 +15,13 @@ def body
         private
 
         def absolute_url(relative_url)
-          URI.join(url, relative_url) unless relative_url.to_s.empty?
+          unless relative_url.to_s.empty?
+            absolute_url = URI.join(url, URI.encode(URI.decode(relative_url)
+                              .gsub('[', '%5B')
+                              .gsub(']', '%5D')))
+                              .to_s
+            CGI.unescape(absolute_url)
```

Comment:
This is quite a convoluted implementation. When given "/person[123]", you're calling `decode` on it (which will have no effect, because `URI.decode` doesn't think that square brackets are special), then running the substitutions on it to get "/person%5B123%5D", which you're then (double-)encoding, to end up with "/person%255B123%255D". Then, because of this, you're having to `CGI.unescape` that at the end (this step shouldn't be necessary at all.)

The issue here is that the URI.decode/encode dance doesn't correctly handle '[', so you need to post-process those, *after* the dance — not in the middle of it.